File: plant.pl
# external, plants seed

if ($player{'inven'}=~/Ei/) {
  ($slot,$item)=split(/\-/, $form{'j'});

  if ($slot<24) {
    if (length($item)==2) {
      if (substr($player{'inven'},$slot*10,2) eq $item) {
        
        if ($item eq "Fa") { $plant="Ia"; $estamp=60; $new="Za00000000"; }
        if ($item eq "Fb") { $plant="Ia"; $estamp=60; $new="Fa".substr($player{'inven'},($slot*10)+2,8); }
        if ($item eq "Fc") { $plant="Ia"; $estamp=60; $new="Fb".substr($player{'inven'},($slot*10)+2,8); }
        if ($item eq "Fd") { $plant="Ia"; $estamp=60; $new="Fc".substr($player{'inven'},($slot*10)+2,8); }
      
        if ($plant) {
          if (!-d "$datadir/dynamic/$player{'tmap'}") { mkdir "$datadir/dynamic/$player{'tmap'}"; }
          if (!-w "$datadir/dynamic/$player{'tmap'}") { chmod 0777, "$datadir/dynamic/$player{'tmap'}" }
          $estamp=$cstamp+$estamp;
          $estamp=sprintf("%08x", $estamp);
          open (FILE,">$datadir/dynamic/$player{'tmap'}/$plant $estamp $player{'tz'}.txt");
          print FILE "\n";
          close FILE;
          substr($player{'inven'},$slot*10,10)="$new";
          $player{'inven'}=~s/Ei/Bd/;
          do 'inv.pl';#print "inv=$player{'inven'}\n";
          print "pop=Planted!\n";
        }
      }
    }
  }
} else {
  print "pop=Need Water!\n";
}

1;

File: src/plant.js
export function plant(){
  if(player.inven.indexOf("Ei")<0)return print+="pop=Need Water!\n";
  var tileset,plant,
  slotitem=form.j.split("-");
  if(slotitem[0]<NumInven&&slotitem[1].length===2&&player.inven.substr(slotitem[0]*=10,2)===slotitem[1]){
    tileset=loadmap(player.tmap),
    g=tileset.substr(player.tz*2,1);
    if(g==="F"||g==="G"){
      if(plant={
        Fa:["Ia",60,"Za"],
        Fb:["Ia",60,"Fa"],
        Fc:["Ia",60,"Fd"],
        Fd:["Ia",60,"Fc"]
      }[slotitem[1]]){
        savedynamic(player.tmap,plant[0],percent0_x(8,plant[1]+cstamp),player.tz);
        player.inven=player.inven.substr(0,slotitem[0])+plant[2]+("Za"===plant[2]?"00000000":player.inven.substr(slotitem[0]+2,8))+player.inven.substr(slotitem[0]+10);
        player.inven.replace(/Ei/,"Bd");
        inv();
        print+="pop=Planted\n";
      }
    }else print+="pop=Not Here\n";
  }
}
